### PR TITLE
BVAL-498 Mandating retrieval of parameter names from class files if p…

### PR DESCRIPTION
**Pull request by gunnarmorling**
…resent

A simple one as it's Friday :)

### CODE REVIEW

File: sources/constraint-declaration-validation.asciidoc [hunk 1]
```diff
@@ -876,7 +876,12 @@ public interface ParameterNameProvider {
 }
 ----
 
-[tck-testable]#A conforming Bean Validation implementation provides a default [classname]`ParameterNameProvider` implementation which returns parameter names in the form arg``PARAMETER_INDEX``, where `PARAMETER_INDEX` starts at 0 for the first parameter, e.g. [varname]`arg0`, [varname]`arg1` etc.#
+[tck-testable]#A conforming Bean Validation implementation provides a default [classname]`ParameterNameProvider` implementation which returns parameter names as stored in the class file containing the valided executable, if present. A conforming implementation must either use the Java reflection API or ensure behavioral compatability to using the reflection API in this way:
```
Comment (gsmet) [inline sources/constraint-declaration-validation.asciidoc]: Hmmm, I nearly fall into your trap! Looks like [tck-testable]# does not support being spanned on multiple lines like that. I would have fixed it on merging but I'm not sure of the best way to fix that as in this case, the items in the list are the important part.

Btw, s/compatability/compatibility/ (even if there is an entry in the wiktionary for compatability: https://en.wiktionary.org/wiki/compatability)

Comment (gunnarmorling) [inline sources/constraint-declaration-validation.asciidoc]: Hum, good catch with the mark-up. What would end up in the assertion file right now?

**Comment by gunnarmorling** [comment]
Force-pushed an update to fix the typo.

File: sources/constraint-declaration-validation.asciidoc (hunk 1 above)
Comment (gunnarmorling) [inline sources/constraint-declaration-validation.asciidoc]: @gsmet I've force-pushed an update which marks only the first sentence as "tck-testable". It's not optimal, but the trailing ":" makes clear that there is more to come which is part of the assertion. Lacking a better solution I think it's acceptable. Can you merge it?

Comment (gsmet) [inline sources/constraint-declaration-validation.asciidoc]: FYI, the main issue was that the `[tck-testable]#` appeared in clear in the spec as it wasn't replaced by our custom block processor.

**Comment by gunnarmorling** [comment]
I see. If there is any way to add the list contents to the assertion,
please feel free to change it :)

**Comment by gsmet** [comment]
Fixed a remaining typo and merged! Thanks. I'll work on the TCK adjustments this afternoon.

**Pull request closed without merging**